packages/blockchain-wallet-v4-frontend/src/data/goals/sagas.js

```diff
@@ -85,12 +86,21 @@ export default ({ api }) => {
     // Special case to handle bitcoin bip21 link integration
     const decodedPayload = decodeURIComponent(pathname + search)
     const bip21Payload = bip21.decode(decodedPayload)
-    const { address } = bip21Payload
-    const { amount, message } = bip21Payload.options || {}
-    const data = { address, amount, description: message }
-    yield put(actions.goals.saveGoal('payment', data))
-    yield put(actions.router.push('/wallet'))
-    yield put(actions.alerts.displayInfo(C.PLEASE_LOGIN))
+    // check for BitPay payment protocol
+    if (bip21Payload.options && bip21Payload.options.r) {
+      const { r } = bip21Payload.options || {}
+      const data = { r }
```

Comment:
confusing intent here. need to make these two lines be more succinct

here is a start => `const r = pathOr({}, ['options', 'r'], bip21Payload)`.  but what is line 92 doing?

Reply:
I followed the existing code in the current defineSendBtcGoal to create r
`const { amount, message } = bip21Payload.options || {}
and line 92 creates a new data object with r as a field to be passed into the saveGoal call

Reply:
creates a new data object with the bitpay link r as the field, to be passed into the runPaymentProtocol goal saga

Reply:
add pathOr in latest